src/ts/state/reducer.ts

```diff
@@ -130,6 +130,9 @@ export interface IPluginState<
   // The id of the match the user is currently hovering over –
   // e.g. to display a tooltip.
   hoverId: string | undefined;
+  //The clientRectIndex of the hover mouseover event. 
```

Comment:
I'd suggest a bit more detail here to let nonplussed devs understand our intent, e.g:

```suggestion
  // The index of the clientRect closest to hover mouseover event.
  // Spans can contain multiple clientRects when they're broken across lines.
  // By indicating which clientRect we're closest to, we can position our match
  // popup next to the correct section of the span.
  // See https://developer.mozilla.org/en-US/docs/Web/API/Element/getClientRects.
```